pytorch_lightning/callbacks/model_checkpoint.py

```diff
@@ -178,30 +178,25 @@ def save_checkpoint(self, trainer, pl_module):
         """
         Performs the main logic around saving a checkpoint
         """
-        # only run on main process
-        if trainer.global_rank != 0:
-            return
+        epoch = trainer.current_epoch
+        # track epoch when ckpt was last checked
+        self.epoch_last_check = trainer.current_epoch
```

Comment:
they are always equal?? I think it won't save anything since with the condition below.

Reply:
Sorry, did not mean to move that. Fixed